### ISSUE TITLE
[Feature] Possibility to delegate authentification to a third pary reverse proxy to make SSO happens

### ISSUE THREAD

**Issue by StudioEtrange** [rommapp/romm#1101]
**Is your feature request related to a problem? Please describe.**
Integration of romm behind a third party reverse proxy which take care of authentification

**Describe the solution you'd like**
The goal is to have an option to delegate entirely the authenfication to a third party. In general it is a reverse proxy (traefik nginx, ...) which take care of authentification on a platform where there is hosted several services (calibreweb, sabnzbd, romm,...). The reverse proxy take care of authentification being connected to any authentification system compatible with the reverse proxy (ie LDAP, keycloak instance or any other authentification system...)
And once the auth is completed, the web route reach romm with a special http header (i.e X_ROMM_USER), and romm used it as the current user.
The point of this option would be to not make authentification inside romm application. Considering in this situation that authentification is out of scope romm.
This kind of system seems simpler to implements than other methods and often used on multi services platform to implement an unified sso without modifying each services.

In two points
* disable any authentification system
* rely on an http header to get the current user

**Describe alternatives you've considered**

**Additional context**

**Comment by zurdi15** [comment]
Hi! duplicated from #540 and #580

**Comment by gantoine** [comment]
> accept an extra http header.

which header is this? romm already accepts authenticated requests with both base64-encoded `Basic` tokens, and oauth2 `Bearer` access and refresh tokens. https://github.com/rommapp/romm/wiki/Authentication

**Comment by ursuscamp** [comment]
I believe what he is requesting (and would be great to have) would be Trusted Header SSO: https://www.authelia.com/integration/trusted-header-sso/introduction/

It works roughly like this:

1. When visiting romm.myserver.com, I have "forward auth" setup in my reverse proxy. The reverse proxy checks with my auth provider if I have a current session.
2. If not, initiate a new session with login form. When login is completed, respond to reverse proxy with trusted auth headers like `Remote-User`, `Remote-Groups`, etc.
3. Reverse proxy then continues by sending the request user to room.myserver.com and copying the headers over to RomM.
4. RomM would see those headers and understand they are setup as the trusted headers vetted by the reverse proxy server and use the info in those header to automatically initiate a new session, create a new user, etc.

**Comment by StudioEtrange** [comment]
What I mean is a simple mechanism present in several product behind a reverse proxy :
Which is mandatory when you make a portal with several services and want that the reverse proxy manage all authentification connected to an arbitrary identity provider, and you do not whant that hundred of servcies in your portal take care of the authentification

mechanism :
The reverse proxy take care of authentifcation purpose with its own mechanism and connected to an identity provider.
Once this is done, the HTTP request is redirected to the final target (here romm) with a HTTP header (might be configurable)
i.e X_PROXY_USER

Romm seeing this check if the user exist, and do NOT launch any mechanism purpose and answer to the request
Romm has to trust this HTTP header and accept the HTTP request on behalf the user specified in X_PROXY_USER

To set trust source, Romm should have a configurable IP source to set the reverse proxy IP : to know which host has to be trusted
And Romm should be able to configure the name of the HTTP_header (ie X_PROXY_USER)


example of products implementing this

calibre-web : https://github.com/janeczku/calibre-web
![image](https://github.com/user-attachments/assets/36f4fe0b-7bcf-44a0-a2a9-a132bcc3329b)

organizr : https://github.com/causefx/Organizr
![image](https://github.com/user-attachments/assets/eee02a30-3f0f-4831-8d71-b15fce3ecdc9)

**Comment by StudioEtrange** [comment]
How to make this issue re examined again ?

**Comment by AngellusMortis** [comment]
The project uses Python and Django is _the project_ when it comes to mature Python Web app dev (and it is also what Authentik uses). So here are the docs on the remote authentication in Django:

https://docs.djangoproject.com/en/5.1/howto/auth-remote-user/

This _could_ be a dupe of implementing OIDC support, if you only want to implement one of the two. OIDC is more a more widely used standard, but receiving a username/groups/etc. from an upstream proxy via headers is _a lot_ simpler to implement then implementing OIDC support. And it appears you are going to get the same level of support from proxy auth providers. Authentik supports passing the arbitrary headers for username/groups. It looks like authelia does as well and a quite glance is the keycloak probably supports it.

**Comment by StudioEtrange** [comment]
@zurdi15 ping

**Comment by WirtsLegs** [comment]
> This _could_ be a dupe of implementing OIDC support, if you only want to implement one of the two. OIDC is more a more widely used standard, but receiving a username/groups/etc. from an upstream proxy via headers is _a lot_ simpler to implement then implementing OIDC support. And it appears you are going to get the same level of support from proxy auth providers. Authentik supports passing the arbitrary headers for username/groups. It looks like authelia does as well and a quite glance is the keycloak probably supports it.

Keycloak does not support forward auth, no, however there is a separate project (keycloak auth proxy) that facilitates this

Ideally id love to see direct OIDC support in RomM but support for Remote User ([RFC 3875](https://tools.ietf.org/html/rfc3875#section-4.1.11)) would be sufficient as well for most people's environment

**Comment by acidRain-burns** [comment]
I would like to see full OIDC implementation as well. These should be separate issues if Romm only intends to implement one of them at this time.

**Comment by StudioEtrange** [comment]
This was originally two differents topic. I think nobody said that only one way could be supported.
* romm, i guess, will implement full OIDC
* my issue was to add another simple way to support SSO using larged used method of Trusted Header SSO, in ecosystem hosting several services 

And romm did not really answer to that point. Support of Trusted Header SSO.
